### PR TITLE
Add initial Vue project setup #8

### LINKED ISSUE #8: Create Initial Vue frontend development project

### PR DESCRIPTION
**Pull request by Komediruzecki**
Add all initial app files (Vue 3, Node LTS (20.x.x), Typescript) 
Add vite test suite for easier testing and lint/prettier

Test: Run development build

### CODE REVIEW

File: tabu-app/index.html
Comment (suncica-negra) [inline tabu-app/index.html]: Možda bi ovdje mogli odmah saviti pravi naziv - Tabu 🙂

File: tabu-app/public/favicon.ico
Comment (suncica-negra) [inline tabu-app/public/favicon.ico:1]: Ako u figmi postoji odgovarajući favicon/logo od Tabu-a možda isto ne bi bilo loše odmah ga staviti umjesto ovog.

File: tabu-app/src/App.vue
Comment (suncica-negra) [inline tabu-app/src/App.vue:26]: Umjesto CSSa bilo bi dobro dodati SCSS. Ali to možda bolje prilikom prvog dodavanja 'stvarnih' stilova.

File: tabu-app/src/assets/logo.svg
Comment (suncica-negra) [inline tabu-app/src/assets/logo.svg:1]: Ako u figmi postoji odgovarajući logo od Tabu-a možda isto ne bi bilo loše odmah ga staviti umjesto ovog.

**Review by suncica-negra — COMMENTED** [review]
Generalno bih rekla da je to za početak ok osim prijašnjih komentara. Ima dosta fajlova koje bude trebalo obrisati, ali to možemo prvom prilikom kad se krene raditi nešto konkretnoga 😃

File: tabu-app/src/App.vue
Comment (Komediruzecki) [reply tabu-app/src/App.vue:26]: I was thinking about something like this: https://tailwindcss.com/docs/guides/vite, have you tried this approach?

On previous projects I used SCSS, but most recently styled components in React, it seemed quite capable and easy to scale, however it might be a bit more complex than needed  https://styled-components.com/

File: tabu-app/public/favicon.ico
Comment (ghost) [reply tabu-app/public/favicon.ico:1]: Favicon and logo have been created. Will you update this accordingly?

File: tabu-app/src/assets/logo.svg
Comment (ghost) [reply tabu-app/src/assets/logo.svg:1]: It was created. Can you update this, then?

File: tabu-app/public/favicon.ico
Comment (suncica-negra) [reply tabu-app/public/favicon.ico:1]: I can. Or is it a question for @Komediruzecki ?

File: tabu-app/src/assets/logo.svg
Comment (suncica-negra) [reply tabu-app/src/assets/logo.svg:1]: I can. Or is it a question for @Komediruzecki ?

File: tabu-app/public/favicon.ico
Comment (Komediruzecki) [reply tabu-app/public/favicon.ico:1]: I'll update this before we merge

File: tabu-app/src/App.vue
Comment (suncica-negra) [reply tabu-app/src/App.vue:26]: I wouldn't say I like remembering many classes to add some style, so I don't like Tailwind and similar frameworks. 

Also, why introduce styles via 'harder' JS? I believe that the application will be 'lighter' with as little JS as possible, and that includes more (S)CSS and less JS. 

But if Tailwind and styled-components will be accepted as a basis for the project, I will adapt 🙂

File: tabu-app/index.html
Comment (Komediruzecki) [reply tabu-app/index.html]: Fixed

File: tabu-app/public/favicon.ico
Comment (Komediruzecki) [reply tabu-app/public/favicon.ico:1]: Fixed

File: tabu-app/src/assets/logo.svg
Comment (Komediruzecki) [reply tabu-app/src/assets/logo.svg:1]: Updated logo

File: tabu-app/src/App.vue
Comment (Komediruzecki) [reply tabu-app/src/App.vue:26]: For the start of project it would be best to start small and see what works, so yeah, lets use SCSS, possibly there is also some neat way to handle vendor prefixes? If that does not work, we can consider other approaches

**Comment by ghost** [comment]
Since I don't understand this code, I'll merge the pull request when you two both agree on it. :)

File: tabu-app/src/App.vue
Comment (suncica-negra) [reply tabu-app/src/App.vue:26]: Ok, I agree 🙂 

Yes, there is https://github.com/postcss/autoprefixer that we can add to the project to take care of prefixes.

**Review by suncica-negra — APPROVED** [review]
I would say that it is ok to begin with and that it can merge into master/main 🥳

**Comment by Komediruzecki** [comment]
> I would say that it is ok to begin with and that it can merge into master/main 🥳

Great, @Vatroslav you can merge it.

**Comment by Komediruzecki** [comment]
> Generalno bih rekla da je to za početak ok osim prijašnjih komentara. Ima dosta fajlova koje bude trebalo obrisati, ali to možemo prvom prilikom kad se krene raditi nešto konkretnoga 😃

Great, I've removed most of template files, hopefully this is good for first merge.

**Pull request merged**
